File: BASIC_OF_ALGORITHM/How_to_Solve_N_&_M/NM_5.py
V = [False] * 10
path = [0] * 10
A = 0
def go(index, n, m):
    if index == m:
        for i in range(m):
            print(path[i], end = " ")
        print()
        return
    for i in range(n):
        if V[i]:
            continue
        V[i] = True
        path[index] = A[i]
        go(index + 1, n, m)
        V[i] = False
n, m = input().split()
A = [int(x) for x in input().split()]
print(A)
A.sort()
go(0, int(n), int(m))

File: BASIC_OF_ALGORITHM/How_to_Solve_N_&_M/NM_5.cpp
#include <iostream>
#include <algorithm>
using namespace std;
int path[10];
int A[10];
bool V[10];
void go(int index, int n, int m) {
	if (index == m) {
		for (int i = 0; i < m; i++) {
			cout << path[i] << " ";
		}
		cout << endl;
		return;
	}

	for (int i = 0; i < n; i++) {
		if (V[i]) continue;
		path[index] = A[i];
		V[i] = true;
		go(index + 1, n, m);
		V[i] = false;
	}
}

int main() {
	int N, M;
	cin >> N >> M;
	for (int i = 0; i < N; i++) {
		cin >> A[i];
	}
	sort(A, A + N);
	go(0, N, M);





	return 0;
}